server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MessageFactory.java

```diff
@@ -97,13 +105,21 @@ public Message fromMetaDataWithContent(MetaDataWithContent message) throws Mailb
                 .replyTo(fromAddressList(mimeMessage.getReplyTo()))
                 .size(message.getSize())
                 .date(message.getInternalDateAsZonedDateTime())
-                .textBody(messageContent.getTextBody().orElse(null))
-                .htmlBody(messageContent.getHtmlBody().orElse(null))
+                .textBody(messageContent.getTextBody().orElse(textBody))
```

Comment:
This is some expensive operation.

Please use orElseGet + compute the replacement text in the orElseGet so that I don't compute it for nothing when there is a text body.